presto-main/src/test/java/io/prestosql/sql/analyzer/TestAnalyzer.java

```diff
@@ -253,10 +256,42 @@ public void testHavingReferencesOutputAlias()
     }
 
     @Test
-    public void testWildcardWithInvalidPrefix()
+    public void testSelectAllColumns()
     {
+        // wildcard without FROM
+        assertFails("SELECT *")
+                .hasErrorCode(COLUMN_NOT_FOUND);
+
+        // wildcard with invalid prefix
         assertFails("SELECT foo.* FROM t1")
                 .hasErrorCode(TABLE_NOT_FOUND);
+
+        assertFails("SELECT a.b.c.d.* FROM t1")
+                .hasErrorCode(TABLE_NOT_FOUND);
+
+        // name aliases are not allowed when selecting all columns from a table
+        assertFails("SELECT t.* AS (a, b, c) FROM (VALUES (1, 2, 3), (10, 20, 30), (100, 200, 300)) t")
+                .hasErrorCode(NOT_SUPPORTED);
+
+        // aliases mismatch
+        assertFails("SELECT (1, 2).* AS (a) FROM t1")
+                .hasErrorCode(MISMATCHED_COLUMN_ALIASES);
+
+        // duplicate alias
+        assertFails("SELECT (1, 2).* AS (a, a) FROM t1")
```

Comment:
This should be allowed, just like `SELECT 1 AS a, 2 AS a` is allowed. The ambiguity should be reported when someone tries to de-reference `a`. For instance:

```sql
SELECT a FROM (
    SELECT (1, 2).* AS (a,a) 
    FROM t
) u
```